include/common/mem.h

```diff
@@ -0,0 +1,51 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+#ifndef __LABWC_MEM_H
+#define __LABWC_MEM_H
+
+#include <stdlib.h>
+
+/*
+ * As defined in busybox, weston, etc.
+ * Allocates zero-filled memory; calls exit() on error.
+ * Returns NULL only if (size == 0).
+ */
+void *xzalloc(size_t size);
+
+/*
+ * Type-safe macros in the style of C++ new/new[].
```

Comment:
IMO we should not refer to another existing standard from another language here and instead explain what we are using it for and why.

Reply:
I don't see the harm here, `new` is well-known and what `znew` is modeled after.

Reply:
`new` might not be well-known for everybody. And I'd like to keep the code base reasonably understandable by people completely new to C as well. You can't do a `man new` like you can do a `man calloc`.

Reply:
That's fair. I'll add some more info to the comment.

Reply:
I pushed a new commit, hopefully there's enough info there now for someone not familiar with C++.